src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -348,10 +540,22 @@ object DeployPlugin extends AutoPlugin {
       System.console.readLine()
       ConfigFactory.empty
     }
-    // Write the fully-resolved environment configuration to disk.
-    val targetFile = targetDir / URLEncoder.encode(hostName, "utf-8") / "env.conf"
-    log.info(s"Writing environment config for '$deployEnv' to '${targetFile.getPath}'")
-    IO.write(targetFile, fullEnvConf.root.render(ConfigRenderOptions.concise))
+
+    // Resolve and write config files for all replicas, including their specified overrides.
+    deployConfig.replicasByHost foreach {
+      case (host, replicas) => replicas foreach { replica =>
+        // Encode hostname to reduce chances of naming problems.
```

Comment:
`.` literals are retained. From the docs:

When encoding a String, the following rules apply:

- The alphanumeric characters "a" through "z", "A" through "Z" and "0" through "9" remain the same.
- The special characters ".", "-", "*", and "_" remain the same.
- The space character "   " is converted into a plus sign "+".
- All other characters are unsafe and are first converted into one or more bytes using some encoding scheme. Then each byte is represented by the 3-character string "%xy", where xy is the two-digit hexadecimal representation of the byte. The recommended encoding scheme to use is UTF-8. However, for compatibility reasons, if an encoding is not specified, then the default encoding of the platform is used.